### PR TITLE
Add project scaffold files

### PR DESCRIPTION
**Pull request by nathandentzau**
This adds the project scaffold files. All of this is a collection of new tools we have been testing on various projects.

* The composer.json file includes a list of project dependencies and drupal modules we use on _majority_ of our projects.
* The package.json file includes tools for compilation, validation and testing frontend code.
* Bootstrap is managed by composer through asset-package.org. This is to allow the Drupal Libraries API to have access to the JavaScript files included in bootstrap. `node-sass` looks in the `/web/libraries` and `/node_modules` directories for include paths.
* `.editorconfig` file added from Drupal core
* SCSS linter is added with rulesets agreed upon by @johncionci 
* ES6 linter is added with rulesets from Drupal core.
* PHP linter is added with rulesets from `drupal/coder`
* A `gulpfile.js` is included for compilation, validation and testing frontend languages.
* PHPUnit is configured to test custom Drupal modules.
* A `.travis.yml` file is included as a template for new Drupal 8 projects.
* A `.lando.yml` file is added for local development.
* This project is open source.

Things we need to figure out:
* Lando local development setup (creating the `settings.php` and files directories)
* Upgrade paths

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# Project dependencies
+/node_modules
+/vendor
+
+# Project test results
+/tests/**/results
+
+# Oomph
+/oomph-infrastructure
```
Comment (marcaddeo) [inline .gitignore]: Don't think this should be in here

File: .lando.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  nodejs:
+    type: node:8.9
+    globals:
+      gulp-cli: latest
+    run:
+      - cd $LANDO_MOUNT && npm install
+
+  phpmyadmin:
+    type: phpmyadmin
```
Comment (marcaddeo) [inline .lando.yml]: Don't you need the environment variables here? or does it work properly now

Comment (nathandentzau) [reply .lando.yml]: Works finally! 🙌

Comment (marcaddeo) [reply .lando.yml]: Noice

File: composer.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        }
+    ],
+    "require": {
+        "bower-asset/bootstrap": "^4.0",
+        "composer/installers": "^1.5",
+        "cweagans/composer-patches": "^1.6",
+        "drupal-composer/drupal-scaffold": "^2.4",
+        "drupal-composer/preserve-paths": "^0.1.4",
+        "drupal/admin_toolbar": "^1.23",
+        "drupal/coder": "^8.2",
```
Comment (marcaddeo) [inline composer.json]: What's coder do? Supply the phpcs standards and stuff?

Comment (nathandentzau) [reply composer.json]: Yup it's the Drupal PHPCS coding standard.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "drupal-scaffold",
+  "version": "1.0.0",
+  "description": "A Drupal 8 scaffold project",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/nathandentzau/drupal-scaffold.git"
```
Comment (marcaddeo) [inline package.json]: Update dis

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        "drupal/coffee": "^1.0",
+        "drupal/config_installer": "^1.8",
+        "drupal/config_split": "^1.3",
+        "drupal/core": "^8.5",
+        "drupal/honeypot": "^1.27",
+        "drupal/oomph_paragraphs": "^1.0",
```
Comment (johnpicozzi) [inline composer.json]: I don't think this should be in here. It may not be needed for some projects like Genscape.

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        "drupal/recreate_block_content": "^2.0",
```
Comment (johnpicozzi) [inline composer.json]: Do we want to use https://www.drupal.org/project/fixed_block_content instead?

File: composer.json (hunk 5 above)
Comment (nathandentzau) [reply composer.json]: o_O I thought we were standardizing using paragraphs going forward?

File: composer.json (hunk 6 above)
Comment (nathandentzau) [reply composer.json]: Might be too early to say we're switching to that. I've only exported 2 blocks with that module so far.

File: composer.json (hunk 5 above)
Comment (johnpicozzi) [reply composer.json]: When we need to create flexible content, but for some sites that kind of content may not be needed. It was identified that Genscape didn't need that kind of content entry.

File: composer.json (hunk 6 above)
Comment (johnpicozzi) [reply composer.json]: I would say if it carries over content it's better than recreate block content and we should consider adding it.

File: composer.json (hunk 5 above)
Comment (nathandentzau) [reply composer.json]: Talked to pico offline. We're going to keep it here and remove it on a per-project basis.

**Comment by marcaddeo** [comment]
Ah, those stallions.. look at them, /approved

**Pull request merged**